utils/filesystem.go

```diff
@@ -26,3 +28,33 @@ func GetDiskUsageStats(path string) (dirSize, inodeCount uint64, err error) {
 
 	return dirSize, inodeCount, err
 }
+
+// IsDirectory tests whether the given path exists and is a directory.
+func IsDirectory(path string) error {
+	info, err := os.Stat(path)
+	if err != nil {
+		return err
+	}
+
+	if !info.Mode().IsDir() {
+		// Return a PathError to be consistent with os.Stat().
+		return &os.PathError{
+			Op:   "stat",
+			Path: path,
+			Err:  syscall.ENOTDIR,
+		}
+	}
+
+	return nil
+}
+
+// MustTempDir uses ioutil.TempDir to create a temporary directory
+// with the given prefix.  It panics on any error.
+func MustTempDir(prefix string) string {
+	path, err := ioutil.TempDir("", prefix)
+	if err != nil {
+		panic(err)
```

Comment:
I would recommend to either move this function into the test suite/framework or add non-panicing error handling.

Reply:
Moved it to `TestFramework` and added the cleanup there too.